.github/workflows/ci.yaml

```diff
@@ -0,0 +1,133 @@
+name: CI
+
+on: [push]
+
+jobs:
+  cvc4:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Get CVC4
+      run: |
+        git clone -b eqrange-quant --single-branch https://github.com/GaloisInc/vadd-CVC4.git .
+    - name: Cache CVC4
+      uses: actions/cache@v1
+      with:
+        path: cache
+        key: ${{ matrix.name }}-cvc-${{ github.sha }}
+        # fall back to (latest) previous cache
+        restore-keys: |
+          ${{ matrix.name }}-cvc
+    - name: Get Antlr
+      run: ./contrib/get-antlr-3.4
+    - name: Configure
+      run: CC="ccache clang" CXX="ccache clang++" ./configure.sh production --static --proofs --prefix=$PWD/dist
+    - name: Build
+      working-directory: build
+      run: |
+        export CCACHE_DIR="$PWD/cache/ccache"
+        export CCACHE_COMPRESS=true
+        export CCACHE_MAXSIZE=400M
+        make -j 8
+    - name: Package
+      run: |
+        find build -name cvc4
+        readelf -d build/bin/cvc4
+    - name: cvc4 artifact
+      uses: actions/upload-artifact@v1
+      with:
+        name: cvc4
+        path: build/bin/cvc4
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # NOTE: must list all targets in `include` here to activate them
+        name: ["Linux 8.8.3"]
+        include:
+          - name: Linux 8.8.3
+            ghc-ver: 8.8.3
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        submodules: true
+    - name: Get GHC
+      run: |
+        sudo add-apt-repository ppa:hvr/ghc
+        sudo apt-get update
+        sudo apt-get install cabal-install-3.0 ghc-${{ matrix.ghc-ver }} llvm
+    - name: Cache
+      uses: actions/cache@v1
+      with:
+        path: home/.cabal/store/ghc-${{ matrix.ghc-ver }}
+        key: ${{ matrix.name }}-build-${{ github.sha }}
+        # fall back to (latest) previous cache
+        restore-keys: |
+          ${{ matrix.name }}-build
+    - name: Cabal update
+      run: |
+        export HOME="$PWD/home"
```

Comment:
The reason you're having to continuously export things is you're not using the [workflow commands](https://help.github.com/en/actions/reference/workflow-commands-for-github-actions) that github provides for this.

```yaml
run: |
  echo "::set-env name=HOME::${{ github.workspace }}/home"
  echo "::add-path::/opt/cabal/bin"
  echo "::add-path::/opt/ghc/${{ matrix.ghc-ver }}/bin"
```

(exporting HOME ad-hoc is likely gonna be a lot more fragile... Seems odd that this is necessary. Can this not run in `$HOME`?)

Reply:
Once I get the things I care about working, I'll address the style comments here.  Thanks for providing the syntax.